### PR TITLE
PR for [UX] Migrate existing markdown content to TSX pages #1056

### PR DESCRIPTION
**Pull request by jamelachahbar**
## 🛠️ Description

#1056
PR for migrating existing markdown pages to TSX pages.
Homepage, landing page for tools and tools pages migrated.
Jest testing framework configured and used to create test for Homepage

### Homepage
image

### Landing Page for Tools
image

### Example Tools Page
image

### Jest test results for Home Page
image

### 🔬 How did you test this change?

> - [ ] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [x] 👍 Manually deployed + verified
> - [x] 💪 Unit tests
> - [ ] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [ ] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [ ] ✅ Updated changelog (required for `dev` PRs)
> - [x] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [ ] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [x] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [ ] ❎ Docs not needed (small/internal change)

### CODE REVIEW

**Comment by jamelachahbar** [comment]
I used **React:FC** for the pages but read that the React team and many TypeScript developers now lean towards **regular function components** rather than React.FC. The main reasons include greater flexibility, better control over types, and reduced verbosity. To change this wouldn't take too much effort, but I need advice and this to be reviewed first.  Looking through samples and examples nowadays, I find regular functions is being used: https://react.dev/learn/typescript#typescript-with-react-components

File: src/web/pages/App.tsx [hunk 1]
```diff
@@ -1,27 +1,37 @@
...
+          <Route path="/tools" element={<ToolsLandingPage />} />
+          <Route path="/tools/finops-hubs" element={<FinOpsHubsPage />} />
+          <Route path="/tools/power-bi-reports" element={<PowerBIReportsPage />} />
+          <Route path="/tools/cost-optimization-workbook" element={<CostOptimizationWorkbookPage />} />
+          <Route path="/tools/finops-workbooks" element={<FinOpsWorkbooksPage />} />
+          <Route path="/tools/governance-workbook" element={<GovernanceWorkbookPage />} />
+          <Route path="/tools/azure-optimization-engine" element={<AzureOptimizationEnginePage />} />
+          <Route path="/tools/powershell-module" element={<PowerShellModulePage />} />
+          <Route path="/tools/bicep-registry-modules" element={<BicepRegistryModulesPage />} />
+          <Route path="/tools/open-data" element={<OpenDataPage />} />
```
Comment (flanakin) [inline src/web/pages/App.tsx:30]: Let's get rid of the extra layer, align them with the current paths, and probably sort them alphabetically for maintainability.
```suggestion
          <Route path="/bicep-registry" element={<BicepRegistryModulesPage />} />
          <Route path="/hubs" element={<FinOpsHubsPage />} />
          <Route path="/open-data" element={<OpenDataPage />} />
          <Route path="/optimization-engine" element={<AzureOptimizationEnginePage />} />
          <Route path="/power-bi" element={<PowerBIReportsPage />} />
          <Route path="/powershell" element={<PowerShellModulePage />} />
          <Route path="/workbooks" element={<FinOpsWorkbooksPage />} />
          <Route path="/workbooks/governance" element={<GovernanceWorkbookPage />} />
          <Route path="/workbooks/optimization" element={<CostOptimizationWorkbookPage />} />
```

File: src/web/pages/AzureOptimizationEnginePage.tsx
Comment (flanakin) [inline src/web/pages/AzureOptimizationEnginePage.tsx:1]: Eventually, we'll have management experiences for each of these. Should we move them to subfolders? Or maybe we can do that later when we have a better idea 🤔 I'm fine either way, just thinking out loud.

File: src/web/tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+
+  },
+  "include": ["src"],
   "files": [],
   "references": [
     { "path": "./tsconfig.app.json" },
-    { "path": "./tsconfig.node.json" }
-  ]
+    { "path": "./tsconfig.node.json" },
+
```
Comment (flanakin) [inline src/web/tsconfig.json:21]:
```suggestion
```

**Review by flanakin — APPROVED** [review]

File: src/web/pages/AzureOptimizationEnginePage.tsx
Comment (jamelachahbar) [reply src/web/pages/AzureOptimizationEnginePage.tsx:1]: We can revisit this later. But I'll think about it over the weekend.

File: src/web/pages/App.tsx (hunk 1 above)
Comment (jamelachahbar) [reply src/web/pages/App.tsx:30]: ok

**Review by Ivanmtta — APPROVED** [review]

**Comment by jamelachahbar** [comment]
Adapted the order and url shortlink for pages. Kept the tools landingpage to not break current linke in homepage.
Will adapt as I'm working through getting the ux design into code locally and leveraging fluent2
image

**Pull request merged**